src/prefect/events/filters.py

```diff
@@ -37,6 +45,15 @@ def excludes(self, event: Event) -> bool:
         """Would the given filter exclude this event?"""
         return not self.includes(event)
 
+    def build_where_clauses(
+        self, db: PrefectDBInterface
+    ) -> Sequence["ColumnExpressionArgument[bool]"]:
+        """Convert the criteria to a WHERE clause."""
+        clauses: List["ColumnExpressionArgument[bool]"] = []
+        for filter in self.get_filters():
+            clauses.extend(filter.build_where_clauses(db))
+        return clauses
```

Comment:
I don't think we want to add the SQL generation stuff to the _client_ schemas (the ones in `prefect.events`), but rather to create the parallel a filters schema in `prefect.server.events`.  Otherwise we'll end up making SQLAlchemy and friends a client-side dependency, which it isn't today.

Reply:
Thanks for the catch! I've copied the event filters to the server-side module and removed the SQL generation from the client-side module in https://github.com/PrefectHQ/prefect/pull/12651/commits/d4c395533ac029d9501131d9561f1a72418ac9ed.